File: src/main/java/com/project/webapp/drivingschool/utils/ProcessingStatus.java
package com.project.webapp.drivingschool.utils;

/**
 * Status przetworzenia dokumentów / płatności
 */
public enum ProcessingStatus {

    /**
     * Etap pierwszy:
     * status nieuregulowany (do wykonania)
     */
    TO_COMPLETE,

    /**
     * Etap drugi:
     * zgłoszenie uregulowania
     */
    REQUESTED,

    /**
     * Etap trzeci:
     * status uregulowany (wykonany)
     */
    COMPLETED

}

File: frontend/src/app/utils/processing-status.ts
/**
 * Status przetworzenia dokumentów / płatności
 */
export enum ProcessingStatus {

  TO_COMPLETE,
  REQUESTED,
  COMPLETED

}

export namespace ProcessingStatus {

  export function translate(status: ProcessingStatus): string {
    let translator: string;
    switch (ProcessingStatus[status] as unknown) {
      case ProcessingStatus.TO_COMPLETE:
        translator = "Do uregulowania";
        break;
      case ProcessingStatus.REQUESTED:
        translator = "Zgłoszony do regulacji";
        break;
      case ProcessingStatus.COMPLETED:
        translator = "Uregulowany";
        break;
      default:
        translator = "";
        break;
    }
    return translator;
  }

}
